tsnet/tsnet_test.go

```diff
@@ -818,65 +827,230 @@ func TestUDPConn(t *testing.T) {
+	s1.lb.DebugForceNetmapUpdate()
+
 	metrics1, err := lc1.UserMetrics(ctx)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	// Note that this test will check for two warnings because the health
-	// tracker will have two warnings: one from the testWarnable, added in
-	// this test, and one because we are running the dev/unstable version
-	// of tailscale.
-	want := `# TYPE tailscaled_advertised_routes gauge
-# HELP tailscaled_advertised_routes Number of advertised network routes (e.g. by a subnet router)
-tailscaled_advertised_routes 2
-# TYPE tailscaled_health_messages gauge
-# HELP tailscaled_health_messages Number of health messages broken down by type.
-tailscaled_health_messages{type="warning"} 2
-# TYPE tailscaled_inbound_dropped_packets_total counter
-# HELP tailscaled_inbound_dropped_packets_total Counts the number of dropped packets received by the node from other peers
-# TYPE tailscaled_outbound_dropped_packets_total counter
-# HELP tailscaled_outbound_dropped_packets_total Counts the number of packets dropped while being sent to other peers
-`
+	parsedMetrics, err := parseMetrics(metrics1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	want := []parsedMetric{
+		{Name: "tailscaled_advertised_routes", Type: "GAUGE", Value: 4},
+		{Name: "tailscaled_approved_routes", Type: "GAUGE", Value: 3},
+		// Two warnings are expected, the one set through the test and
+		// one that tells the client that a dev/unstable version is running.
+		{
+			Name:   "tailscaled_health_messages",
+			Type:   "GAUGE",
+			Labels: "type=warning,",
+			Value:  2,
+		},
+		{
+			Name:   "tailscaled_inbound_bytes_total",
+			Type:   "COUNTER",
+			Labels: "path=derp,",
+			Value:  148,
+		},
+		{
+			Name:   "tailscaled_inbound_bytes_total",
+			Type:   "COUNTER",
+			Labels: "path=direct_ipv4,",
+			Value:  380,
+		},
+		{
+			Name:   "tailscaled_inbound_bytes_total",
+			Type:   "COUNTER",
+			Labels: "path=direct_ipv6,",
+		},
+		{
+			Name:   "tailscaled_inbound_packets_total",
+			Type:   "COUNTER",
+			Labels: "path=derp,",
+			Value:  1,
+		},
+		{
+			Name:   "tailscaled_inbound_packets_total",
+			Type:   "COUNTER",
+			Labels: "path=direct_ipv4,",
+			Value:  5,
+		},
+		{
+			Name:   "tailscaled_inbound_packets_total",
+			Type:   "COUNTER",
+			Labels: "path=direct_ipv6,",
+		},
+		{
+			Name:   "tailscaled_outbound_bytes_total",
+			Type:   "COUNTER",
+			Labels: "path=derp,",
+			Value:  148,
+		},
+		{
+			Name:      "tailscaled_outbound_bytes_total",
+			Type:      "COUNTER",
+			Labels:    "path=direct_ipv4,",
+			ValueFrom: 1100,
+			ValueTo:   1300,
+		},
+		{
+			Name:   "tailscaled_outbound_bytes_total",
+			Type:   "COUNTER",
+			Labels: "path=direct_ipv6,",
+		},
+		{
+			Name:   "tailscaled_outbound_packets_total",
+			Type:   "COUNTER",
+			Labels: "path=derp,",
+			Value:  1,
+		},
+		{
+			Name:      "tailscaled_outbound_packets_total",
+			Type:      "COUNTER",
+			Labels:    "path=direct_ipv4,",
+			ValueFrom: 12,
+			ValueTo:   17,
```

Comment:
this all seems way too fragile and too specific for a test

Tests that require golden value rebasing when something unrelated far away changes are a pretty painful maintenance cost.